### PR TITLE
Virology Reworked

### PR DESCRIPTION
**Pull request by TheCaramelion**
## About The Pull Request

Gets rid of Virus 2 and +10 years old untouched code. Updates the entire diseases system and makes the virology role useful for once.

Advance diseases are engineered by virologists, they can range from useful to dangerous. Virologists can use the new chemicals to improve and evolve their viruses. These new viruses can be cured with specific chemicals, as well as the ones made by the virologists. If a virus has been found, you can check the cure with a simple health analyzer, otherwise the PanDEMIC machine will show anything in it. Viruses will be auto-discovered if they reach half or more in it's stages.

## Changelog
:cl:
qol: Rework to all of Virology
add: Many effects for the new disease system
add: Chemicals for the new disease system
/:cl:

### CODE REVIEW

**Comment by Skysteak** [comment]
Has a lot of potential! I don't know how to read code very well, but as stated before (and saying again here for the record), I do think maintaining a self cure chance ('toughing it out') , regardless of inaction would be good, even if non-viable for severe viruses. 

I think it is set like this in the PR already, but as before, ensuring viruses poof upon death is good, and ensures that a station struggling to cope with a death virus in say, an event, doesn't just death spiral. Whilst it may seem authentic to a viral outbreak, what it'd essential translate into is just mass resleeving, manual or automatic.

Just in case: Remember the Virus Immunity trait exists!

**On particular virus symptoms:**

-Coughing making you drop small items makes me feel conflicted , but I understand the need for it to actually be disruptive without being fatal. I think it'll come down to the rate it procs (in terms of how it 'feels' in game).

-The Fever Temp Raise; This is once again 'I read code bad' as a question, but will it potentially endanger teshari or people with the cold adapted trait? 

-The Magnetism; Is it just moving silicons about or physically throwing them? Do note that if it's the latter, that A. Borg stuns are a lot more obnoxious than person ones, due to it clearing your hotbar  every time it happens, and B. Whilst armour helps, being thrown into walls is quite nasty, even for silicons. Also I don't think you can do this, but it'd be fun if synths were also affected by it too!

**Comment by TheCaramelion** [comment]
Most of the viruses are able to be self-healed, there's even a virus effect that clears other virus effects.
Diseases already clear upon death, and I wasn't planning on making any new that would actually work on dead people unless it's for some kind of event or PoI where corpses CAN transmit them.
Virus immunity stays the same and won't infect people, however, it might make asyntomatic carriers.

While all the effects are still being worked on, the ones that were pointed out:
Coughing only works for small items like pens, cups and papers, not much that can be solved with a small click
Fever shouldn't go into lethal temperatures for species that can't handle the heat, it's capped at their max
Magnetism does throw silicons, but the effects won't happen if you're not in sight of the patient with the virus. It won't throw people against walls. Most of the time it would end up in accidental vore

And there will be an affect that will make synthetics be affected, so they can also get the good viruses that a virologist may create

**Comment by TheCaramelion** [comment]
Needs fixing on Soluna Nexus but the rest should be good to go

File: modular_chomp/code/datums/diseases/advance/symptoms/flesh_eating.dm [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+/datum/symptom/flesh_eating/Activate(datum/disease/advance/A)
+	..()
+	if(prob(SYMPTOM_ACTIVATION_PROB))
+		var/mob/living/M = A.affected_mob
+		switch(A.stage)
+			if(2,3)
+				to_chat(M, span_warning("[pick("You feel a sudden pain across your body.", "Drops of blood appear suddenly on your skin.")]"))
+			if(4,5)
+				to_chat(M, span_userdanger("[pick("You cringe as a violent pain takes over your body.", "It feels like your body is eating itself inside out.", "IT HURTS.")]"))
```
Comment (Kashargul) [inline modular_chomp/code/datums/diseases/advance/symptoms/flesh_eating.dm:35]: spans are defined as
#define span_type(str) ("<span class='type'>" + str + "</span>") so they can be chained and even procs can be used in them. There's no need to use any with "[proc("")]"

Counts for all uses here. The only important is that the function returns only strings

**Review by Kashargul — CHANGES_REQUESTED** [review]

File: modular_chomp/code/datums/diseases/advance/symptoms/hallucigen.dm [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+/datum/symptom/hallucigen/Activate(datum/disease/advance/A)
+	..()
+	if(prob(SYMPTOM_ACTIVATION_PROB))
+		var/mob/living/carbon/M = A.affected_mob
+		switch(A.stage)
+			if(1, 2)
+				to_chat(M, span_warning(pick("Something appears in your peripheral vision, then winks out.", "You hear a faint whisper with no source.", "Your head aches.")))
+			if(3, 4)
+				to_chat(M, span_warning("<b>[pick("Something is following you.", "You are being watched.", "You hear a whisper in your ear.", "Thumping footsteps slam toward you from nowhere.")]</b>"))
```
Comment (Kashargul) [inline modular_chomp/code/datums/diseases/advance/symptoms/hallucigen.dm:35]: span_boldwarning()

File: modular_chomp/code/datums/diseases/advance/symptoms/spin.dm [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+	level = 1
+	severity = 1
+	var/list/directions = list(2,4,1,8,2,4,1,8,2,4,1,8,2,4,1,8,2,4,1,8)
+
+/datum/symptom/mlem/Activate(var/datum/disease/advance/A)
+	..()
+
+	if(prob(SYMPTOM_ACTIVATION_PROB))
+		if(A.affected_mob.buckled())
+			to_chat(viewers(A.affected_mob), span_warning("[A.affected_mob.name] struggles violently against their restraints!"))
```
Comment (Kashargul) [inline modular_chomp/code/datums/diseases/advance/symptoms/spin.dm:34]: Can to_chat handle more than one target? It wants 1 usually

File: modular_chomp/code/datums/diseases/advance/symptoms/spin.dm [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+		else
+			to_chat(viewers(A.affected_mob), span_warning("[A.affected_mob.name] spins around violently!"))
```
Comment (Kashargul) [inline modular_chomp/code/datums/diseases/advance/symptoms/spin.dm:36]: Can to_chat handle more than one target? It wants 1 usually

File: modular_chomp/code/datums/diseases/advance/symptoms/weigh.dm [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+
+/datum/symptom/weight_loss/Activate(datum/disease/advance/A)
+	..()
+	if(prob(SYMPTOM_ACTIVATION_PROB))
+		var/mob/living/M = A.affected_mob
+		switch(A.stage)
+			if(1, 2, 3, 4)
+				to_chat(M, span_warning(pick("You feel hungry.", "You crave for food.")))
+			else
+				to_chat(M, span_warning("<i>[pick("So hungry...", "You'd kill someone for a bite of food...", "Hunger cramps seize you...")]</i>"))
```
Comment (Kashargul) [inline modular_chomp/code/datums/diseases/advance/symptoms/weigh.dm:36]: span_italics

File: modular_chomp/code/datums/diseases/food_poisoning.dm [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+				affected_mob.emote("groan")
+			if(prob(1))
+				to_chat(affected_mob, span_danger("Your stomach hurts."))
+			if(prob(1))
+				to_chat(affected_mob, span_danger("You feel sick."))
+			if(prob(5))
+				if(affected_mob.nutrition > 10)
+					affected_mob.emote("vomit")
+				else
+					to_chat(affected_mob, "Your stomach lurches painfully")
```
Comment (Kashargul) [inline modular_chomp/code/datums/diseases/food_poisoning.dm:64]: No sorting span

File: modular_chomp/code/game/machinery/pandemic.dm [hunk 7]
```diff
@@ -0,0 +1,377 @@
...
+		if(stat & (NOPOWER|BROKEN))
+			return
+		if(beaker)
+			to_chat(user, span_warning("A beaker is already loaded into the machine!"))
+			return
+
+		user.drop_item()
+		beaker = I
+		beaker.loc = src
+		to_chat(usr, span_notice("You add the beaker to the machine."))
```
Comment (Kashargul) [inline modular_chomp/code/game/machinery/pandemic.dm:373]: usr? why not user

File: modular_chomp/code/datums/diseases/advance/symptoms/spin.dm (hunk 3 above)
Comment (Kashargul) [reply modular_chomp/code/datums/diseases/advance/symptoms/spin.dm:34]: handled in the queue of the subsystem

**Comment by TheCaramelion** [comment]
Upstream asked for this, upporting

**Pull request closed without merging**